nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -57,3 +71,324 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
+                    self.update_cve(matching_dlc_cve, cve_info)
+                    continue
+
+            # API Rest Timer
+            sleep(6)
+
+        self.logger.info(
+            "Performed discovery on all software. Created %s CVE.", cve_counter, extra={"grouping": "CVE Creation"}
+        )
+        self.session.close()
+
+    def associate_software_to_cve(self, software_id, cve_id):
+        """A function to associate software to a CVE."""
+        cve = CVELCM.objects.get(id=cve_id)
+        software = SoftwareVersion.objects.get(id=software_id)
+
+        try:
+            cve.affected_softwares.add(software)
+
+        except IntegrityError as err:
+            self.logger.error(
+                "Unable to create association between CVE and Software Version.  ERROR: %s",
+                err,
+                extra={"object": cve, "grouping": "CVE Association"},
+            )
+
+    @staticmethod
+    def create_cpe_software_search_urls(vendor: str, platform: str, version: str) -> list:
+        """Uses netutils.platform_mapper to construct proper search URLs.
+
+        Args:
+            vendor (str): Software Vendor (Examples: Cisco, Juniper, Arista)
+            platform (str): Software Platform (Examples: IOS, JunOS, EOS)
+            version (str): Software Version
+
+        Returns:
+            list: List of URLS that associated CVEs may be found
+        """
+        cpe_urls = get_nist_vendor_platform_urls(vendor, platform, version)
+
+        return cpe_urls
+
+    def create_dlc_cves(self, cpe_cves: dict) -> None:
+        """Create the list of needed items and insert into to DLC CVEs."""
+        created_count = 0
+
+        for cve, info in cpe_cves.items():
+            try:
+                description = (
+                    f"{info['description'][0:251]}..." if len(info["description"]) > 255 else info["description"]
+                )
+            except TypeError:
+                description = "No Description Provided from NIST DB."
+
+            create_cves, created = CVELCM.objects.get_or_create(  # pylint: disable=unused-variable
+                name=cve,
+                description=description,
+                published_date=date.fromisoformat(info.get("published_date", "1900-01-01")[0:10]),
+                last_modified_date=date.fromisoformat(info.get("modified_date", "1900-01-01")[0:10]),
+                link=info["url"],
+                cvss=info["cvss_base_score"],
+                severity=info["cvss_severity"],
+                cvss_v2=info["cvssv2_score"],
+                cvss_v3=info["cvssv3_score"],
+                comments="ENTRY CREATED BY NAUTOBOT NIST JOB",
+            )
+
+            if created:
+                created_count += 1
+
+        self.logger.info("Created New CVEs.", extra={"grouping": "CVE Creation"})
+
+    def get_cve_info(self, cpe_software_search_urls: list, software_id=None) -> dict:
```

Comment:
We already have SoftwareVersion object in the calling method so it's probably better to use that instead of passing `software_id` around.